framework/TSA/Wavelet.py

```diff
@@ -0,0 +1,142 @@
+# Copyright 2017 Battelle Energy Alliance, LLC
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+# http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+"""
+  Discrete Wavelet Transformation for Characterization of Time-Series Data.
+"""
+import numpy as np
+
+from utils import InputData, InputTypes, xmlUtils
+from .TimeSeriesAnalyzer import TimeSeriesAnalyzer
+
+
+# utility methods
+class Wavelet(TimeSeriesAnalyzer):
+  """
+    Perform Discrete Wavelet Transformation on time-dependent data.
+  """
+
+  @classmethod
+  def getInputSpecification(cls):
+    """
+      Method to get a reference to a class that specifies the input data for
+      class cls.
+      @ Out, inputSpecification, InputData.ParameterInput, class to use for
+        specifying input of cls.
+    """
+    import pywt
+    specs = super(Wavelet, cls).getInputSpecification()
+    specs.name = 'Wavelet'
+    specs.descriiption = """TimeSeriesAnalysis algorithm """
+    specs.addSub(InputData.parameterInputFactory(
+      'family',
+      contentType=InputTypes.StringType,
+      descr="""The type of wavelet to use for the transformation. Possible values are:
+      haar, db1, db2, db3, db4, db5, db6,
+      db7, db8, db9, db10, db11, db12, db13, db14, db15, db16, db17, db18, db19,
+      db20, db21, db22, db23, db24, db25, db26, db27, db28, db29, db30, db31, db32,
+      db33, db34, db35, db36, db37, db38 sym2, sym3, sym4, sym5, sym6, sym7, sym8,
+      sym9, sym10, sym11, sym12, sym13, sym14, sym15, sym16, sym17, sym18, sym19,
+      sym20, coif1, coif2, coif3, coif4, coif5, coif6, coif7, coif8, coif9, coif10,
+      coif11, coif12, coif13, coif14, coif15, coif16, coif17, bior1.1, bior1.3,
+      bior1.5, bior2.2, bior2.4, bior2.6, bior2.8, bior3.1, bior3.3, bior3.5,
+      bior3.7, bior3.9, bior4.4, bior5.5, bior6.8 rbio1.1, rbio1.3, rbio1.5,
+      rbio2.2, rbio2.4, rbio2.6, rbio2.8, rbio3.1, rbio3.3, rbio3.5, rbio3.7,
+      rbio3.9, rbio4.4, rbio5.5, rbio6.8, dmey, gaus1, gaus2, gaus3, gaus4, gaus5,
+      gaus6, gaus7, gaus8, mexh, morl, cgau1, cgau2, cgau3, cgau4, cgau5, cgau6,
+      cgau7, cgau8, shan, fbsp, cmor"""
+    ))
```

Comment:
Can we provide a reference here to let the reader find out which type to use? These designations mean very little to me at first glance.

Reply:
This reference should probably also be in the tex file, I just realized. Thanks.